### PR TITLE
Allow admins to fail jobs without container

### PR DESCRIPTION
**Pull request by fhammerl**

### CODE REVIEW

File: src/Runner.Worker/JobExtension.cs [hunk 1]
```diff
@@ -672,5 +674,13 @@ private async Task CheckDiskSpaceAsync(IExecutionContext context, CancellationTo
             Trace.Info($"Total accessible running process: {snapshot.Count}.");
             return snapshot;
         }
+
+        private void ValidateJobContainer(JobContainer container)
+        {
+            if (StringUtil.ConvertToBoolean(Environment.GetEnvironmentVariable(Constants.Variables.Actions.FailJobsWithoutJobContainer)) && container == null)
+            {
+                throw new ArgumentException("Jobs without a jobContainer are forbidden on this machine because of the security policy set by the self-hosted runner administrator.");
```
Comment (fhammerl) [inline src/Runner.Worker/JobExtension.cs:682]: Verbose exception instead of ArgUtil.NotNull

File: src/Runner.Common/Constants.cs [hunk 2]
```diff
@@ -227,6 +227,7 @@ public static class Actions
                 //
                 public static readonly string AllowUnsupportedCommands = "ACTIONS_ALLOW_UNSECURE_COMMANDS";
                 public static readonly string AllowUnsupportedStopCommandTokens = "ACTIONS_ALLOW_UNSECURE_STOPCOMMAND_TOKENS";
+                public static readonly string FailJobsWithoutJobContainer = "ACTIONS_FAIL_JOBS_WITHOUT_JOBCONTAINER";
```
Comment (thboop) [inline src/Runner.Common/Constants.cs:230]:
```suggestion
                public static readonly string FailJobsWithoutJobContainer = "ACTIONS_RUNNER_REQUIRE_JOB_CONTAINER";
```

File: src/Runner.Worker/JobExtension.cs [hunk 3]
```diff
@@ -672,5 +674,13 @@ private async Task CheckDiskSpaceAsync(IExecutionContext context, CancellationTo
             Trace.Info($"Total accessible running process: {snapshot.Count}.");
             return snapshot;
         }
+
+        private static void ValidateJobContainer(JobContainer container)
+        {
+            if (StringUtil.ConvertToBoolean(Environment.GetEnvironmentVariable(Constants.Variables.Actions.FailJobsWithoutJobContainer)) && container == null)
+            {
+                throw new ArgumentException("Jobs without a jobContainer are forbidden on this machine because of the security policy set by the self-hosted runner administrator.");
```
Comment (thboop) [inline src/Runner.Worker/JobExtension.cs:682]:
```suggestion
                throw new ArgumentException("Jobs without a job container are forbidden on this self hosted runner, please add a container to your job or contact your self-hosted runner administrator.");
```

File: src/Runner.Common/Constants.cs [hunk 4]
```diff
@@ -227,6 +227,7 @@ public static class Actions
                 //
                 public static readonly string AllowUnsupportedCommands = "ACTIONS_ALLOW_UNSECURE_COMMANDS";
                 public static readonly string AllowUnsupportedStopCommandTokens = "ACTIONS_ALLOW_UNSECURE_STOPCOMMAND_TOKENS";
+                public static readonly string FailJobsWithoutJobContainer = "ACTIONS_RUNNER_REQUIRE_JOB_CONTAINER";
```
Comment (thboop) [inline src/Runner.Common/Constants.cs:230]: should we rename this variable as well?

File: src/Runner.Worker/JobExtension.cs [hunk 5]
```diff
@@ -672,5 +674,13 @@ private async Task CheckDiskSpaceAsync(IExecutionContext context, CancellationTo
             Trace.Info($"Total accessible running process: {snapshot.Count}.");
             return snapshot;
         }
+
+        private static void ValidateJobContainer(JobContainer container)
+        {
+            if (StringUtil.ConvertToBoolean(Environment.GetEnvironmentVariable(Constants.Variables.Actions.FailJobsWithoutJobContainer)) && container == null)
+            {
+                throw new ArgumentException("Jobs without a job container are forbidden on this self hosted runner, please add a 'container:' to your job or contact your self-hosted runner administrator.");
```
Comment (thboop) [inline src/Runner.Worker/JobExtension.cs:682]:
```suggestion
                throw new ArgumentException("Jobs without a job container are forbidden on this  runner, please add a 'container:' to your job or contact your self-hosted runner administrator.");
```

**Review by thboop — APPROVED** [review]
lgtm minor nits

**Comment by fhammerl** [comment]
Updated with updated var names and an L0

File: src/Runner.Common/Constants.cs [hunk 6]
```diff
@@ -227,6 +227,7 @@ public static class Actions
                 //
                 public static readonly string AllowUnsupportedCommands = "ACTIONS_ALLOW_UNSECURE_COMMANDS";
                 public static readonly string AllowUnsupportedStopCommandTokens = "ACTIONS_ALLOW_UNSECURE_STOPCOMMAND_TOKENS";
+                public static readonly string RequireJobContainer = "ACTIONS_REQUIRE_JOB_CONTAINER";
```
Comment (thboop) [inline src/Runner.Common/Constants.cs:230]: I have `ACTIONS_RUNNER_REQUIRE_JOB_CONTAINER` in my ADR, do you have a preference for this? Looks like we use a mix of both in the runner

File: src/Runner.Worker/JobExtension.cs [hunk 7]
```diff
@@ -672,5 +674,13 @@ private async Task CheckDiskSpaceAsync(IExecutionContext context, CancellationTo
             Trace.Info($"Total accessible running process: {snapshot.Count}.");
             return snapshot;
         }
+
+        private static void ValidateJobContainer(JobContainer container)
+        {
+            if (StringUtil.ConvertToBoolean(Environment.GetEnvironmentVariable(Constants.Variables.Actions.RequireJobContainer)) && container == null)
+            {
+                throw new ArgumentException("Jobs without a job container are forbidden on this  runner, please add a 'container:' to your job or contact your self-hosted runner administrator.");
```
Comment (thboop) [inline src/Runner.Worker/JobExtension.cs:682]:
```suggestion
                throw new ArgumentException("Jobs without a job container are forbidden on this runner, please add a 'container:' to your job or contact your self-hosted runner administrator.");
```

File: src/Test/L0/Worker/JobExtensionL0.cs [hunk 8]
```diff
@@ -211,6 +211,24 @@ public async Task JobExtensionBuildPreStepsList()
             }
         }
 
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public async Task JobExtensionBuildFailsWithoutContainerIfRequired()
+        {
+            Environment.SetEnvironmentVariable(Constants.Variables.Actions.RequireJobContainer, "1");
```
Comment (thboop) [inline src/Test/L0/Worker/JobExtensionL0.cs:219]:
```suggestion
            Environment.SetEnvironmentVariable(Constants.Variables.Actions.RequireJobContainer, "true");
```

**Review by thboop — APPROVED** [review]
lgtm

**Pull request merged**